src/graph/graphQL.ts

```diff
@@ -72,15 +77,34 @@ export function prettyPrintErrors(errors: GraphQLError[], query?: string): strin
 /**
  * Resolve and read the contents of a GrapqQL query or subscription file
  * @param {string} path
+ * @param {string} current
+ * @param {{[p: string]: string | boolean | number}} parameters
  * @returns {string}
  */
-export function resolveAndReadFileSync(path: string, current: string = appRoot.path): string {
+export function resolveAndReadFileSync(path: string,
+                                       current: string = appRoot.path,
+                                       parameters: {
+                                           [name: string]: string | boolean | number;
+                                       }): string {
     if (!path.endsWith(".graphql")) {
         path = `${path}.graphql`;
     }
     const absolutePath = p.resolve(current, path);
     if (fs.existsSync(absolutePath)) {
-        return fs.readFileSync(absolutePath).toString();
+        let content = fs.readFileSync(absolutePath).toString();
+        for (const key in parameters) {
+            if (parameters.hasOwnProperty(key)) {
+                const value: any = parameters[key];
+                if (typeof value === "string") {
+                    content = content.replace(`$${key}`, `"${value}"`);
```

Comment:
This will only replace the first occurrence of `$VARIABLE`.

Reply:
And will also match the variable name as a substring. For example, if you had a variable named `$foo` and a second named `$fooBar`, performing the replacement on `$foo` will match `$fooBar` should that come first in the string.

Reply:
Odd, you're right although
```
/**
      * Replaces text in a string, using a regular expression or search string.
      * @param searchValue A string to search for.
      * @param replaceValue A string containing the text to replace for every successful match of searchValue in this string.
      */
```